sql/catalyst/src/main/java/org/apache/spark/sql/execution/UnsafeExternalRowIterator.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution;
+
+import org.apache.spark.sql.catalyst.expressions.UnsafeRow;
+import scala.collection.AbstractIterator;
+
+public abstract class UnsafeExternalRowIterator extends AbstractIterator<UnsafeRow> {
```

Comment:
This should maybe extend Closeable?

Reply:
... and now you can just copy in the implementation that was previously in the anonymous subclass below? that might be cleaner overall. Hm, but maybe it won't work out because it has to call cleanupResources(). That part may have to stay in an anonymous subclass.

Reply:
Done